roles/provision_vcenter/tasks/main.yml

```diff
@@ -3,26 +3,17 @@
   ansible.builtin.include_tasks: validate_inputs.yml
 
 - name: Check If VCenter Already Exists
-  community.vmware.vmware_guest_info:
+  vmware.vmware.guest_info:
     hostname: "{{ provision_vcenter_hostname }}"
     username: "{{ provision_vcenter_username }}"
     password: "{{ provision_vcenter_password }}"
     port: "{{ provision_vcenter_port | default(omit) }}"
     validate_certs: "{{ provision_vcenter_validate_certs | default(omit) }}"
-    datacenter: "{{ provision_vcenter_vc_target_datacenter | default('') }}"
     name: "{{ provision_vcenter_vm_name }}"
   register: _vcenter_lookup
-  failed_when: false
```

Comment:
IIUC this line handles a failure in case `"{{ provision_vcenter_vm_name }}"` doesn't exist. I think it's better to leave it as is.

Reply:
Never mind, I found `fail_on_missing=False`